tests/TestRegistryTest.cpp

```diff
@@ -323,3 +323,36 @@ TEST(TestRegistry, ResetPluginsWorks)
     myRegistry->resetPlugins();
     LONGS_EQUAL(0, myRegistry->countPlugins());
 }
+
+TEST(TestRegistry, listGroupNames)
+{
+    test1->setGroupName("GROUP_1");
+    myRegistry->addTest(test1);
+    test2->setGroupName("GROUP_2");
+    myRegistry->addTest(test2);
+    test3->setGroupName("GROUP_1");
+    myRegistry->addTest(test3);
+
+    myRegistry->listTestGroupNames(*result);
+    SimpleString s = output->getOutput();
+    STRCMP_EQUAL("GROUP_1 GROUP_2 ", s.asCharString());
```

Comment:
Better. Now, I see there is the extra space at the end. Do we want to do something about that? I assume it can lead to nasty surprises when people use this functionality?